opencart 1.5.6.4/upload/admin/view/template/module/smaily_for_opencart.tpl

```diff
@@ -114,6 +129,64 @@
             </tr>  
           </table>
         </div>
+        <!-- Customer sync -->
+        <div id="tab-sync">
+          <table class="form">
+            <tr>
+              <td><?php echo $entry_enable_subscriber_title; ?></td>
+              <td><select name="smaily_for_opencart_enable_subscribe" id="input-subscriber-status" class="form-control">
+                    <?php if ($subscribe_status) { ?>
+                    <option value="1" selected="selected"><?php echo $text_enabled; ?></option>
+                    <option value="0"><?php echo $text_disabled; ?></option>
+                    <?php } else { ?>
+                    <option value="1"><?php echo $text_enabled; ?></option>
+                    <option value="0" selected="selected"><?php echo $text_disabled; ?></option>
+                    <?php } ?>    
+                  </select>
+              </td>
+            </tr>
+            <tr>
+              <td><?php echo $entry_customer_sync_fields_title; ?></td>
+              <td><select class="form-control" name="smaily_for_opencart_syncronize_additional[]" id="customer_sync_fields" multiple="multiple">
+                    <?php
+                      // All available options
+                      $sync_options = [
+                        'firstname'  => $firstname,
+                        'lastname'  => $lastname,
+                        'telephone'  => $telephone,
+                        'date_added' => $date_added
+                      ];
+                      // Add options for select.
+                      foreach ($sync_options as $value => $name) {
+                        $selected = is_array($syncronize_additional) && in_array($value, $syncronize_additional) ? 'selected' : '';
+                        echo("<option value='$value' $selected>$name</option>");
```

Comment:
Try to keep templates (or HTML) just as a data representation layer (avoid having business logic in templates when possible).

Rendering this select can be refactored to:
```php
<select class="form-control" name="smaily_for_opencart_syncronize_additional[]" id="customer_sync_fields" multiple="multiple">
  <?php foreach ($sync_options as $value => $sync_field) { ?>
  <option
    value="<?php echo $value; ?>"
    <?php if ($sync_field['selected') { ?> selected=""<?php } ?>>
    <?php echo $sync_field['label']; ?>
  </option>
  <?php } ?>
</select>
```

Where `$sync_options` is something like:
```php
$sync_options = array(
  'firstname' => array(
    'label' => $this->language->get('firstname'),
    'selected' => false,
  ),
  'lastname' => array(
    'label' => $this->language->get('lastname'),
    'selected' => false,
  ),
  'telephone' => array(
    'label' => $this->language->get('telephone'),
    'selected' => false,
  ),
  'date_added' => array(
    'label' => $this->language->get('date_added'),
    'selected' => false,
  ),
);

foreach ($this->data['subscribe_additional'] as $field) {
  $sync_options[$field]['selected'] = true;
}
```